lib/avro/src/main/kotlin/kafkasnoop/avro/AvroSchema.kt

```diff
@@ -42,5 +43,31 @@ data class AvroSchema(val fullName: String, val needs: List<String>, val schema:
             }
             return AvroSchema("$ns.$name", needs, schema)
         }
+
+        private fun extractNonPrimitiveTypes(elem: JsonElement?): List<String> {
+
+            if (elem == null) {
+                return emptyList()
+            }
+
+            return if (elem.isJsonObject) {
+                extractNonPrimitiveTypes(elem.asJsonObject.get("type"))
+                extractNonPrimitiveTypes(elem.asJsonObject.get("items"))
+            } else if (elem.isJsonArray) {
+                elem.asJsonArray.filter {
+                    if (it.isJsonPrimitive) {
```

Comment:
your can simplify this with `.filter { it.isJsonPrimitive && it.asString !in primitive }`